src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/KeyValuePairListEncoding.cs

```diff
@@ -23,13 +23,13 @@ internal static class KeyValuePairListEncoding
             }
         }
 
-        static IEnumerable<string> ReadEntries(string rawText)
+        static IEnumerable<string> ReadEntries(string rawText, char separator)
         {
             bool escaped = false;
             int entryStart = 0;
             for (int i = 0; i < rawText.Length; i++)
             {
-                if (rawText[i] == ',' && !escaped)
+                if (separator == rawText[i] && !escaped)
```

Comment:
nit, but it's more idiomatic in C# to keep the original ordering. I guess I'd formalise it as the thing that's changing in the loop is on the LHS, and the value to compare against on the RHS.